### PR TITLE
More Bugfixes client and serverside

### PR DESCRIPTION
**Pull request by Ch0wW**
This PR does the following:
- Fix the incorrect `level.killed_monsters` value when connecting to a server.
- Fix a broadcasted message on CTF with an incorrect parameter.
- Do not interpolate the camera during the teleporting freeze.
- Correct `sv_monstersrespawn` not working midgame.
- Correct a rare occurence of monsters disappearing randomly after changing the difficulty from nightmare to a lower one.
- Fix powerups being carried after callvoting a map.
- (Serverside only) Respawned monsters on Nightmare or from an archvile properly increases the `level.total_monsters` value.
- Remove duplicated messages when a client has RCON access.
- Forbid clients from changing compatibility values clientside in the compability menu.
- Removes unused CVARs clientside and serverside.

### CODE REVIEW

File: common/p_user.cpp [hunk 1]
```diff
@@ -117,6 +117,17 @@ void P_ClearPlayerCards(player_t& p)
...
 
+/**
+ * @brief Clear all powerups from a player.
+ *
+ * @param p Player to clear.
+ */
+void P_ClearPlayerPowerups(player_t& p)
+{
+	for (size_t i = 0; i < NUMPOWERS; i++)
+		p.powers[i] = false;
```
Comment (AlexMax) [inline common/p_user.cpp:128]: `powers` is not a boolean, you should probably prefer 0.

File: client/src/m_options.cpp [hunk 2]
```diff
@@ -563,23 +567,23 @@ menu_t SoundMenu = {
...
+	{svdiscrete, "Items can drop off ledges",      {&co_allowdropoff},      {2.0}, {0.0}, {0.0}, {OnOff}},
+	{redtext,   " ",								{NULL},                  {0.0}, {0.0}, {0.0}, {NULL}},
+	{bricktext, "Engine Compatibility",				{NULL},                  {0.0}, {0.0}, {0.0}, {NULL}},
+	{svdiscrete, "BOOM actor/sector/line checks",  {&co_boomphys},			 {2.0}, {0.0}, {0.0}, {OnOff}},
+	{svdiscrete, "ZDOOM 1.23 physics",             {&co_zdoomphys},         {2.0}, {0.0}, {0.0}, {OnOff}},
+	{redtext,   " ",								{NULL},                  {0.0}, {0.0}, {0.0}, {NULL}},
+	{bricktext, "Sound",							{NULL},                  {0.0}, {0.0}, {0.0}, {NULL}},
+	{svdiscrete, "Fix silent west spawns",         {&co_nosilentspawns},    {2.0}, {0.0}, {0.0}, {OnOff}},
+	{svdiscrete, "ZDoom Sound Response",			{&co_zdoomsound},		 {2.0}, {0.0}, {0.0}, {OnOff}},
+    {svdiscrete, "Global Pickup Sounds",			{&co_globalsound},		 {2.0}, {0.0}, {0.0}, {OnOff}},
```
Comment (AlexMax) [inline client/src/m_options.cpp:586]: This section has a lot of spacing changes - any reason why?

**Review by AlexMax — CHANGES_REQUESTED** [review]
Just a few changes and questions.

File: common/p_user.cpp (hunk 1 above)
Comment (Ch0wW) [reply common/p_user.cpp:128]: I adapted with what existed already - will change it now.

File: client/src/m_options.cpp (hunk 2 above)
Comment (Ch0wW) [reply client/src/m_options.cpp:586]: I'd believe it was due to a cause of bad formatting. I'll try to restore it to a normal state.

**Review by AlexMax — APPROVED** [review]
LGTM

**Pull request merged**